services/token.rb

```diff
@@ -0,0 +1,30 @@
+require "httparty"
+require "json"
+require "base64"
+
+module Services
+  class Token
+    include HTTParty
+
+    base_uri "https://accounts.spotify.com"
+    def self.auth_string_calc
+      # client_id = "50283f807bc74fecbc2e61563d34e051"
+      # client_secret = "37afc91e05f146dba6b3dcca57b8b900"
+      # auth_string = "Basic #{Base64.encode64(client_id)}:#{Base64.encode64(client_secret)}"
```

Comment:
i think this code could be better  instead of the next line when the cliente  key secret change maybe But your logic is ok ;)